modules/installation-osp-converting-ignition-resources.adoc

```diff
@@ -0,0 +1,171 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_openstack/installing-openstack-user.adoc
+
+[id="installation-osp-converting-ignition-resources_{context}"]
+= Preparing the bootstrap Ignition file
+
+The bootstrap Ignition file that the installation program generates is too large to pass to {rh-openstack} directly.
+
+First, edit and upload the bootstrap Ignition configuration file. Then, create a small Ignition file ("shim") to pass to OpenStack Compute (Nova). The shim instructs the server to download the large bootstrap Ignition file.
+
+.Prerequisites
+
+* The bootstrap Ignition file that the installation program generates (`bootstrap.ign`)
+* The infrastructure ID from the installation program's metadata file is set as an environment variable (`$INFRA_ID`)
+** If the variable is not set, see *Creating the Kubernetes manifest and Ignition config files*.
+* An HTTP(S)-accessible way to store the bootstrap ignition file
+** The documented procedure uses the OpenStack Image service (Glance) or OpenStack Storage service (Swift), but you could also use the Amazon S3, an internal HTTP server, or an ad hoc Nova server
+
+.Procedure
+
+. Edit the bootstrap Ignition file to update the hostname file when it runs. You can do this manually, but it is faster to run the following Python script:
++
+[source,python]
+----
+import base64
+import json
+import os
+
+with open('bootstrap.ign', 'r') as f:
+    ignition = json.load(f)
+
+files = ignition['storage'].get('files', [])
+
+infra_id = os.environ.get('INFRA_ID', 'openshift').encode()
+hostname_b64 = base64.standard_b64encode(infra_id + b'-bootstrap\n').decode().strip()
+files.append(
+{
+    'path': '/etc/hostname',
+    'mode': 420,
+    'contents': {
+        'source': 'data:text/plain;charset=utf-8;base64,' + hostname_b64,
+        'verification': {}
+    },
+    'filesystem': 'root',
+})
+
+ignition['storage']['files'] = files;
```

Comment:
FYI we will need to backport [this](https://github.com/openshift/installer/pull/3290) to 4.4.

When installing on a self-signed-certificate-secured OpenStack cluster, the Ignition-shim file must carry the certificate so that the machine can download the real Ignition through the self-signed-cert HTTPS connection.

CC @mandre

Reply:
https://github.com/openshift/openshift-docs/pull/20091/commits/00e45f553b3a097ee1b3c2272332d26a758abf1f